hal/armv7a/arch/cpu.h

```diff
@@ -49,9 +49,15 @@
 #define SYSTICK_INTERVAL 1000
 
 
+#define STACK_ALIGN 8
+
+
+#define STACK_SIZE(t) ((sizeof(t) + 3u) & ~0x3u)
+
+
 #define PUTONSTACK(kstack, t, v) \
 	do { \
-		(kstack) -= (sizeof(t) + 3) & ~0x3; \
+		(kstack) = (char *)(kstack) - STACK_SIZE(t); \
```

Comment:
To avoid the warning - perhaps we should always pass `kstack` casted to `char *` to avoid UB on the other hand? Or leave this line as is, but fix usage where `kstack` is `u32`. I guess the second option would be better
```suggestion
		(kstack) -= STACK_SIZE(t); \
```